x/leverage/types/string.go

```diff
@@ -0,0 +1,68 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (ap *AccountPosition) String() string {
+	s := "special:\n"
+	for _, wsp := range ap.specialPairs {
+		s += fmt.Sprintf("  %s\n", wsp)
+	}
+	s += "normal:\n"
+	for _, wnp := range ap.normalPairs {
+		s += fmt.Sprintf("  %s\n", wnp)
+	}
+	for _, sc := range ap.unpairedCollateral {
+		s += fmt.Sprintf("  [%s, -]\n", sc)
+	}
+	for _, sb := range ap.unpairedBorrows {
+		s += fmt.Sprintf("  [-, %s]\n", sb)
+	}
+	return s
+}
+
+// String represents a WeightedNormalPair in the form [WeightedDecCoin, WeightedDecCoin]
+// e.g. [10 uumee @ 0.35, 3.5 uumee @ 0.35]
+func (wnp WeightedNormalPair) String() string {
+	return fmt.Sprintf("[%s, %s]", wnp.Collateral, wnp.Borrow)
+}
+
+// String represents a WeightedSpecialPair in the form [DecCoin, DecCoin] @ weight
+// e.g. [10 uumee, 3.5 uumee] @ 0.35
+func (wsp WeightedSpecialPair) String() string {
+	return "[" + sDecCoin(wsp.Collateral) + ", " + sDecCoin(wsp.Borrow) + "] @ " + sDec(wsp.SpecialWeight)
+}
+
+// String represents a WeightedDecCoin in the form coin @ weight
+// e.g. 10 uumee @ 0.35
+func (wdc WeightedDecCoin) String() string {
+	return sDecCoin(wdc.Asset) + " @ " + sDec(wdc.Weight)
+}
+
+// sDec formats a sdk.Dec as a string with no trailing zeroes after the decimal point,
+// omitting the decimal point as well for whole numbers
+func sDec(d sdk.Dec) string {
```

Comment:
optional, and might want to move to `util/string` package if they ever need to be exported.